### PR TITLE
BaseMVAValueMapProducer configuration and parameter validation

### PR DESCRIPTION
**Pull request by vlimant**
#### PR description:

the original implementation of `BaseMVAValueMapProducer` turns out to be quite cumbersome ; this PR makes it slimmer and hopefully less error-prone

#### PR validation:

nano matrix

tagging @arizzi

### CODE REVIEW

**Comment by vlimant** [comment]
enable nano

File: PhysicsTools/NanoAOD/python/jetsAK4_CHS_cff.py [hunk 1]
```diff
@@ -246,44 +246,55 @@
...
+        cms.PSet( name = cms.string("Jet_leptonPtRel"), expr = cms.string("userFloat('leptonPtRelv0')")),
+        cms.PSet( name = cms.string("Jet_leptonDeltaR"), expr = cms.string("userFloat('leptonDeltaR')")),
+        cms.PSet( name = cms.string("Jet_neHEF"), expr = cms.string("neutralHadronEnergyFraction()")),
+        cms.PSet( name = cms.string("Jet_neEmEF"), expr = cms.string("neutralEmEnergyFraction()")),
+        cms.PSet( name = cms.string("Jet_vtxPt"), expr = cms.string("userFloat('vtxPt')")),
+        cms.PSet( name = cms.string("Jet_vtxMass"), expr = cms.string("userFloat('vtxMass')")),
+        cms.PSet( name = cms.string("Jet_vtx3dL"), expr = cms.string("userFloat('vtx3dL')")),
+        cms.PSet( name = cms.string("Jet_vtxNtrk"), expr = cms.string("userInt('vtxNtrk')")),
+        cms.PSet( name = cms.string("Jet_vtx3deL"), expr = cms.string("userFloat('vtx3deL')")),
+        #cms.PSet( name = cms.string("Jet_numDaughters_pt03"), expr = cms.string("")),
```
Comment (vlimant) [inline PhysicsTools/NanoAOD/python/jetsAK4_CHS_cff.py:264]: @nurfikri89 : can you please check why those variables were present in the ordering, but not in the variable definition ? what am I missing ?

**Comment by vlimant** [comment]
type muon

**Comment by vlimant** [comment]
type ego

**Comment by vlimant** [comment]
type jetmet

**Comment by vlimant** [comment]
type egamma

**Comment by vlimant** [comment]
please test

**Comment by jfernan2** [comment]
enable nano

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by antoniovilela** [comment]
+1

**Pull request merged**